### PR TITLE
[Installer] Automatically detect conflicts between framework names.

### PR DESCRIPTION
**Pull request by segiddins**
Closes https://github.com/CocoaPods/CocoaPods/issues/2943.

\c @neonichu @kylef

### CODE REVIEW

File: lib/cocoapods/installer.rb [hunk 1]
```diff
@@ -327,6 +328,20 @@ def determine_dependency_product_types
...
+    def verify_no_duplicate_framework_names
+      aggregate_targets.each do |aggregate_target|
+        aggregate_target.user_build_configurations.keys.each do |config|
+          pod_targets = aggregate_target.pod_targets_for_build_configuration(config)
+          vendored_frameworks = pod_targets.flat_map(&:file_accessors).flat_map(&:vendored_frameworks)
+          frameworks = vendored_frameworks.map { |fw| fw.basename('.framework') }
+          frameworks += pod_targets.select { |pt| pt.should_build? && pt.requires_frameworks? }.map(&:product_module_name)
+
+          duplicates = frameworks.group_by { |f| f }.select { |_, v| v.size > 1 }.keys
+          raise "The '#{aggregate_target.label}' target has frameworks with conflicting names: #{duplicates.to_sentence}." unless duplicates.empty?
```
Comment (kylef) [inline lib/cocoapods/installer.rb]: Should this be an Informative?

Comment (segiddins) [inline lib/cocoapods/installer.rb]: Yup, good catch! Will do once I finish up this next PR.

File: lib/cocoapods/installer.rb [hunk 2]
```diff
@@ -90,6 +90,7 @@ def install!
       resolve_dependencies
       download_dependencies
       determine_dependency_product_types
+      verify_no_duplicate_framework_names
```
Comment (kylef) [inline lib/cocoapods/installer.rb]: Should this only verify if the integration type is frameworks?

Comment (segiddins) [inline lib/cocoapods/installer.rb]: No, in theory there could be conflicts in `vendored_frameworks`

File: lib/cocoapods/installer.rb (hunk 1 above)
Comment (segiddins) [inline lib/cocoapods/installer.rb]: Done

**Pull request merged**